aws/src/integration/java/org/apache/iceberg/aws/lakeformation/TestLakeFormationDML.java

```diff
@@ -0,0 +1,159 @@
+
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
+import org.junit.Test;
+import software.amazon.awssdk.services.glue.model.AccessDeniedException;
+import software.amazon.awssdk.services.lakeformation.model.Permission;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+public class TestLakeFormationDML extends LakeFormationTestBase {
+
+  @Test
+  public void testLoadTableWithNoTableAccess() {
+    String testDbName = getRandomDbName();
+    String testTableName = getRandomTableName();
+    lfRegisterPathRoleCreateDb(testDbName);
+    lfRegisterPathRoleCreateTable(testDbName, testTableName);
+    try {
+      AssertHelpers.assertThrows("attempt to load a table without SELECT permission should fail",
+          AccessDeniedException.class,
+          "Insufficient Lake Formation permission(s)",
+          () -> glueCatalogPrivilegedRole.loadTable(TableIdentifier.of(Namespace.of(testDbName), testTableName)));
+    } finally {
+      lfRegisterPathRoleDeleteTable(testDbName, testTableName);
+      lfRegisterPathRoleDeleteDb(testDbName);
+    }
+  }
+
+  @Test
+  public void testLoadTableSuccess() {
+    String testDbName = getRandomDbName();
+    String testTableName = getRandomTableName();
+    lfRegisterPathRoleCreateDb(testDbName);
+    lfRegisterPathRoleCreateTable(testDbName, testTableName);
+    grantTablePrivileges(testDbName, testTableName, Permission.SELECT);
+    try {
+      glueCatalogPrivilegedRole.loadTable(TableIdentifier.of(Namespace.of(testDbName), testTableName));
+    } finally {
+      lfRegisterPathRoleDeleteTable(testDbName, testTableName);
+      lfRegisterPathRoleDeleteDb(testDbName);
+    }
+  }
+
+  @Test
+  public void testUpdateTableWithNoInsertAccess() {
+    String testDbName = getRandomDbName();
+    String testTableName = getRandomTableName();
+    lfRegisterPathRoleCreateDb(testDbName);
+    lfRegisterPathRoleCreateTable(testDbName, testTableName);
+    grantTablePrivileges(testDbName, testTableName, Permission.SELECT);
+    try {
+      Table table = glueCatalogPrivilegedRole.loadTable(TableIdentifier.of(Namespace.of(testDbName), testTableName));
+      DataFile dataFile = DataFiles.builder(partitionSpec)
+          .withPath("/path/to/data-a.parquet")
+          .withFileSizeInBytes(10)
+          .withRecordCount(1)
+          .build();
+      AssertHelpers.assertThrows("attempt to insert to a table without INSERT permission should fail",
+          S3Exception.class,
+          "Access Denied",
+          () -> table.newAppend().appendFile(dataFile).commit());
+    } finally {
+      lfRegisterPathRoleDeleteTable(testDbName, testTableName);
+      lfRegisterPathRoleDeleteDb(testDbName);
+    }
```

Comment:
[suggestion] try structuring the code in BeforeEach, AfterEach 

      lfRegisterPathRoleDeleteTable(testDbName, testTableName);
      lfRegisterPathRoleDeleteDb(testDbName);

seems to be good candidate for AfterEach, 

    lfRegisterPathRoleCreateDb(testDbName);
    lfRegisterPathRoleCreateTable(testDbName, testTableName);

seems to good candidate for BeforeEach

can make testDbName / testTableName as class property which you set before each ut execution and similarly you can reference it AfterEach to get rid of it.